Toolchain/Stubs/README.md

```diff
@@ -1,21 +1,34 @@
 # Library stubs
 
-This directory contains stubs for SerenityOS libraries (LibC, LibM, LibDl, LibPthread)
-that are referenced from the LLVM runtime libraries. These are needed by the linker
-in order to add the required `DT_NEEDED` entries.
+This directory contains stubs for SerenityOS's LibC that are referenced from the LLVM runtime
+libraries. These are needed by the linker in order to add the required `DT_NEEDED` entries and to
+not emit errors regarding undefiend libc symbols. Additionally, it provides fake empty libunwind.so
+and libc++.so for CMake configuration checks to succeed when bootstrapping the OS.
 
 ## Do these need to be updated?
 
-Generally, no. LLVM does not use the header files to decide which functionality it can
-use. After adding a new function to a header, you don't have to worry about LLVM
-toolchain builds failing because the symbol is not present in the stubs.
+Most likely no but it depends. Either way, if you are reading this, you are probably qualified
+enough to figure out if a failing LLVM toolchain configuration or build is caused by an out-of-date
+LibC stub.
 
-## How to generate these?
+## How to generate LibC stub?
 
-First, you need to have a working SerenityOS installation that's been built by the
-Clang toolchain. Then, using the `llvm-ifs` tool, these libraries need to be converted
-into a stripped-down stub form. To do that, run the following command:
+First, you need to compile the LLVM toolchain and the SerenityOS's LibC. This will be a bit awkward
+(but possible with enough time spend modifying CMake files) until (unless) we solve the dependency
```

Comment:
wanna explain briefly what to mess with in the cmake files? not a requirement, but this reads like "fuck you, not my problem".

Reply:
It is really just "I don't care how you get your libc, just get it". I _think_ what has to be done is the following:
- compile clang without runtimes
- make llvm not set -Wl,-z,defs for runtime
- make empty libc.so stub
- compile runtime
- compile Serenity
- generate proper libc.so stub
- undo -Wl,-z,defs patch

Or you can cheat and first compile Serenity using GCC toolchain and grab libc from there.

Reply:
(I mostly meant giving a better description in the document than ":shrug: get a libc build")